### PR TITLE
SuperDoc: Collaboration updates

### PR DESCRIPTION
**Pull request by harbournick**
- Adds all collaboration infra to superdoc
- Tweaks to editor.js for collaboration service use (in node)

### CODE REVIEW

File: packages/super-editor/src/style.css [hunk 1]
```diff
@@ -63,7 +63,33 @@
   border: 1px solid #1355FF !important;
 }
 
-/* li,ol,ul {
-  margin: 0;
-  padding: 0;
-} */
+/* Collaboration cursors */
```
Comment (artem-harbour) [inline packages/super-editor/src/style.css]: I would probably prefer to keep these styles here.
`packages/super-editor/src/core/config/style.js`

But our styles are not very standardized and require some revision in any case.

File: packages/super-editor/src/extensions/document-attributes/document-attributes.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { Node, Attribute } from '@core/index.js';
+
+export const DocumentAttributes = Node.create({
```
Comment (artem-harbour) [inline packages/super-editor/src/extensions/document-attributes/document-attributes.js]: What is this node used for?

File: packages/super-editor/src/components/SuperEditor.vue [hunk 3]
```diff
@@ -36,34 +37,73 @@ const props = defineProps({
...
+  const metaMap = ydoc.getMap('meta');
 
-  const [content, media] = await Editor.loadXmlData(props.fileSource);
+  const checkData = () => {
+    const docx = metaMap.get('docx');
+    if (docx) {
+      initEditor(docx);
+    } else if (retries > 0) {
+      console.debug(`Waiting for 'docx' data... retries left: ${retries}`);
+      setTimeout(checkData, interval); // Retry after the interval
```
Comment (artem-harbour) [inline packages/super-editor/src/components/SuperEditor.vue]: Should we call `clearTimeout` on component unmount?

Comment (harbournick) [reply packages/super-editor/src/components/SuperEditor.vue]: Ah yes, good call. Adding that now.

File: packages/super-editor/src/extensions/document-attributes/document-attributes.js (hunk 2 above)
Comment (harbournick) [reply packages/super-editor/src/extensions/document-attributes/document-attributes.js]: Actually I'll remove it - ended up using default values for this instead. 
The reason I added initially was that the 'document' node is a special case and can't have attrs associated with it (or at least, they are not retained in collaboration), so we were losing the <w:document> tags' attrs. I figured I'd sync it with its own hidden node, but using default values here seems to work fine instead and its simpler.

**Pull request merged**